### PR TITLE
Expose downsampleRaw for testing

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This pr exposes `downsampleRaw` for testing purpose.

### CODE REVIEW

File: pkg/compact/downsample/downsample.go [hunk 1]
```diff
@@ -100,7 +102,7 @@ func Downsample(
 
 	var (
 		aggrChunks []*AggrChunk
-		all        []sample
+		all        []tsdbutil.Sample
```
Comment (bwplotka) [inline pkg/compact/downsample/downsample.go:105]: So in theory this struct was supposed to be used only for testing. It's controversial, but the idea is that we wanted to have reproduced sample struct everywhere as it's easy to copy such small struct and easier to type it (as you could see when refactoring).

Indeed it makes it's difficult to reuse though... I am not sure if we want to use Prometheus struct designed for testing though in our production code 🤔

Comment (yeya24) [reply pkg/compact/downsample/downsample.go:105]: Got it. I see. Then what about exposing `downsample.sample` struct and use it?

Comment (yeya24) [reply pkg/compact/downsample/downsample.go]: Updated with the way mentioned in the OH.

**Review by kakkoyun — APPROVED** [review]
LGTM 🚀

**Comment by yeya24** [comment]
Will merge in 1h if there is no objection.

**Review by bwplotka — APPROVED** [review]
Thanks! LGTM 💪🏽

**Pull request merged**